app/forms/teacher_interface/upload_form.rb

```diff
@@ -0,0 +1,68 @@
+class TeacherInterface::UploadForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveStorageValidations
+
+  attr_accessor :document
+
+  attribute :original_attachment
+  attribute :translated_attachment
+
+  validates :document, presence: true
+
+  validates :original_attachment,
+            content_type: %i[png jpg jpeg pdf doc docx],
+            size: {
+              less_than: 50.megabytes
+            },
+            allow_nil: true
+
+  validates :translated_attachment,
+            content_type: %i[png jpg jpeg pdf doc docx],
+            size: {
+              less_than: 50.megabytes
+            },
+            allow_nil: true
+
+  def present?
+    original_attachment.present? || translated_attachment.present?
+  end
+
+  def save
+    return false unless valid?
+    return false if blank?
+
+    if original_attachment.present?
+      document.uploads.create!(
+        attachment: original_attachment,
+        translation: false
+      )
+    end
+
+    if translated_attachment.present?
+      document.uploads.create!(
+        attachment: translated_attachment,
+        translation: true
+      )
+    end
+
+    true
+  end
+end
+
+# The following is required as we're using a validation library which is
+# usually used directly on a model and received an ActiveStorage::Blob
+# rather than an ActionDispatch::Http::UploadedFile.
+class ActionDispatch::Http::UploadedFile
+  def attached?
+    true
+  end
+
+  def blob
+    self
+  end
+
+  def byte_size
+    size
+  end
+end
```

Comment:
I'd like to get this PR in before the user research tomorrow, but I'm going to write a card to change this as I don't like it.

I think it's better, since the validation we want is very specific, to not use the library and instead build our own basic validation on the content type and file size so we con't need to worry about making a `ActionDispatch::Http::UploadedFile` instance look like a `ActiveStorage::Blob` instance.